engine/common/provider/engine.go

```diff
@@ -163,16 +168,33 @@ func (e *Engine) onEntityRequest(originID flow.Identifier, req *messages.EntityR
 	// try to retrieve each entity and skip missing ones
 	entities := make([]flow.Entity, 0, len(req.EntityIDs))
 	entityIDs := make([]flow.Identifier, 0, len(req.EntityIDs))
+	seen := make(map[flow.Identifier]struct{})
 	for _, entityID := range req.EntityIDs {
+		// skip requesting duplicate entity IDs
+		if _, ok := seen[entityID]; ok {
+			e.log.Debug().
```

Comment:
any idea if this happens with the node software as it is today? I think we should return an error in this case and flag this as suspected malicious behavior

Reply:
why? wouldn't we want to ignore/fail any suspected malicious behavior instead of continuing on a best effort basis? It seems to me this would either indicate there's a bug in the implementation, or the peer is attempting something malicious.

Reply:
Yes your right since this is only possible during an attempt to do something malicious